packages/destination-actions/src/destinations/engage-messaging-sendgrid/sendEmail/index.ts

```diff
@@ -26,6 +26,56 @@ const insertEmailPreviewText = (html: string, previewText: string): string => {
   return $.html()
 }
 
+const insertUnsubscribeLinks = (
+  html: string,
+  emailProfile: any,
+  spaceId: string,
+  statsClient: StatsClient | undefined,
+  tags: string[],
+  groupId?: string,
+  logger?: Logger | undefined
+): string => {
+  const globalUnsubscribeLink = emailProfile?.unsubscribeLink
+  const preferencesLink = emailProfile?.preferencesLink
+  const unsubscribeLinkRef = 'a[href*="[upa_unsubscribe_link]"]'
+  const preferencesLinkRef = 'a[href*="[upa_preferences_link]"]'
+  const $ = cheerio.load(html)
+  if (groupId && groupId != '') {
+    const group = emailProfile?.groups.find((group: { id: string }) => group?.id === groupId)
+    const groupUnsubscribeLink = group?.groupUnsubscribeLink
+    $(unsubscribeLinkRef).each(function () {
+      logger?.info(`TE Messaging: Email Group Unsubscribe link replaced  - ${spaceId} ${groupId}`)
+      statsClient?.incr('actions-personas-messaging-sendgrid.replaced_group_unsubscribe_link', 1, tags)
+      $(this).attr('href', groupUnsubscribeLink)
+    })
+  } else {
+    $(unsubscribeLinkRef).each(function () {
+      logger?.info(`TE Messaging: Email Global Unsubscribe link replaced  - ${spaceId}`)
+      statsClient?.incr('actions-personas-messaging-sendgrid.replaced_global_unsubscribe_link', 1, tags)
+      $(this).attr('href', globalUnsubscribeLink)
+    })
+  }
+  $(preferencesLinkRef).each(function () {
+    logger?.info(`TE Messaging: Email Preferences link replaced  - ${spaceId}`)
+    statsClient?.incr('actions-personas-messaging-sendgrid.replaced_preferences_link', 1, tags)
+    if (!preferencesLink || preferencesLink == '') {
```

Comment:
Empty string is a falsy value. So, it should be okay to do just do the below here.
```suggestion
    if (!preferencesLink) {
```